app/models/session_activation.rb

```diff
@@ -67,10 +56,6 @@ def exclusive(id)
 
   private
 
-  def assign_user_agent
```

Comment:
Side note here - I'm not totally sure what the assign_user_agent before_save callback on SessionActivation is doing. The column has a default of "" and is not null (and was like that when method added), so it seems like this value would be set at the db save level if not sooner. When I remove the callback nothing fails. In this PR I've converted it to a `normalizes` call in the concern, which seems conceptually similar if not exactly the same as the callback. It's possible it could just be removed.

Reply:
Ah, nevermind on this! Full CI run revealed issue.